src/git/commit.ml

```diff
@@ -33,7 +33,7 @@ sig
   module D: S.DECODER
     with type t = t
      and type init = Cstruct.t
-     and type error = [ `Decoder of string ]
+     and type error = Error.Decoder.t0
```

Comment:
`t0` is not a very good error name. I understand the separation in "levels" but the result is not so great.